### PR TITLE
Please check homework 3 and merge into master branch

### PR DESCRIPTION
**Pull request by spiridonovfed**

### CODE REVIEW

File: homework3/task01.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    # '2'
+"""
+
+
+def cache(times):
+    def the_real_decorator(function):
+        def wrapper(*args):
+            nonlocal times
+            if args in memory and times:
+                times -= 1
```
Comment (fake-commit) [inline homework3/task01.py:36]: This solution will cache results only for the first input, the next inputs won't be cached.

File: homework3/task04.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Write a function that detects if a number is Armstrong number in functionaly style:
+ - use map or other utilities from functools library,
+ - use anonymous functions (or use function as argument)
+ - do not use loops, preferably using list comprehensions
+
+### Example function signature and call
+"""
+
+
+def is_armstrong(number: int) -> bool:
```
Comment (fake-commit) [inline homework3/task04.py:21]: Add an extra check if `number` is negative then function should return False (now it fails with ValueError)

File: tests/homework3/test_task04.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+def test_negative_case2():
+    assert not is_armstrong(11)
+
+
+def test_positive_case3():
+    assert is_armstrong(407)
+
+
```
Comment (fake-commit) [inline tests/homework3/test_task04.py:23]: Please add one test for negative number.

**Review by fake-commit — CHANGES_REQUESTED** [review]

**Review by fake-commit — APPROVED** [review]
